src/debugger/DebuggerTcp.cpp

```diff
@@ -563,12 +569,38 @@ bool DebuggerTcp::receive(uint8_t* buffer, size_t& length)
     return true;
 }
 
-void DebuggerTcp::close(void)
+void DebuggerTcp::close(CloseReason reason)
 {
-    if (enabled()) {
-        tcpCloseSocket(m_socket);
-        disable();
+    if (!enabled()) {
+        return;
+    }
+
+    if (reason != CloseAbortConnection) {
+        uint8_t message[4];
+        uint16_t reasonID;
+
+        switch (reason) {
+        default:
+            reasonID = 1000;
+            break;
```

Comment:
minor proposal,
How about initializing `reasonID` as 1000 instead of adding default case in switch?

Other looks good :)

Reply:
That was my idea, but compiler complained about unhandled enums in switch, so I needed to add a default case. If you prefer I can move the assignment above (still need to add a `default: break;`).

Reply:
I see, then what about replacing this switch statement by if-elseif statement?

Reply:
Good idea. Since the standard defines only a few errors, it is unlikely to add many more cases.